src/librustc_apfloat/lib.rs

```diff
@@ -380,7 +380,9 @@ pub trait Float
     fn from_bits(input: u128) -> Self;
     fn from_i128_r(input: i128, round: Round) -> StatusAnd<Self> {
         if input < 0 {
-            Self::from_u128_r(-input as u128, -round).map(|r| -r)
+            // This is equivalent to `(-input) as u128` except it doesn't overflow on i128::MIN
+            let abs_input = !(input as u128) + 1;
```

Comment:
Ehm, just use `wrapping_neg`? Somehow I forgot to.

Reply:
Oh, right, that's a thing.